### ISSUE TITLE
IRC support

### ISSUE THREAD

**Issue by hannesm** [hannesm/jackline#104]
would be great... now that multi-user chat is there... maybe using https://github.com/johnelse/ocaml-irc-client is sensible?

**Comment by lukasepple** [comment]
You mean IRC support within a jabber client?

I don't know if this is beyond scope. On the other hand that would mean hassle-free OTR for IRC queries…

**Comment by hannesm** [comment]
well, apart from XMPP I do use IRC (and am not too happy with my current client), and certainly abstracting over the underlying protocol would help to cleanup code..

**Comment by lukasepple** [comment]
Why not then?

**Comment by cfcs** [comment]
Hm. I thought everyone was supposed to move to XMPP and just use IRC bridges/transports? :-(

On the other hand I don't see people moving away from IRC any time soon, so why not.
Perhaps we should move to the clean React style first, or in the process of implementing this.
IRC would also require more UI things, so having the new UI lib in place before spending time on this would also be good.

**Comment by lukasepple** [comment]
I don't see why anybody would quit using IRC. It does not require an account and MUC is a hassle, also there's no such thing as bouncers or anything for XMPP MUC.

**Comment by hannesm** [comment]
@cfcs yep, definitively something for the rather far future (first, get the XMPP support up to speed)...

@lukasepple you don't need bouncers since XMPP provides server-side logging which a client can request..

**Comment by lukasepple** [comment]
@hannesm all of the backlog? Found that feature implemented rather poorly in xmpp clients.

**Comment by hannesm** [comment]
I don't know server implementations, but at least in the "specification" there's no limit https://xmpp.org/extensions/xep-0045.html#enter-managehistory

client implementation, such as jackline, are currently poor (jackline specifically because I rarely use multi-user chat and am not aware about which server supports which strategy; and what a user interface should be (automated pulling, manual request via command, ...) [unfortunately also currently busy with other things]
